force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -26,184 +26,164 @@
 
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
+        for (Integer i = 0; i < str.length(); i ++) {
+            String currentChar = str.substring(i, i + 1);
+            if (currentChar == 'a' || currentChar == 'e' || currentChar == 'i' || currentChar == 'o' || currentChar == 'u') {
+                result ++;
+            }
         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        return result; 
     }
 
     /**
-     * Question 5
      * Given a list of scientist names, combine their names together with a comma if it contains the letter 'g'.
      * The final string should not end with a comma.
      * Example: findTheScientist() should return "Alan Turing,Grace Hopper,Guido van Rossum,Stephen Hawking"
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
+        String combinedNames = '';
 
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
+        for (String name : scientistNames) {
+            String lowerName = name.toLowerCase();
+            if(!lowerName.contains('g')) {
+                continue;
+                } else {
+                    combinedNames += name + ',';
+                } 
+        }
 
-        return null; // Replace null with the variable you used to store the result
+        combinedNames = combinedNames.removeEnd(',');
+        return combinedNames;
     }
 
     /**
-     * Question 6
      * This method calculates the absolute value of the difference between of two input numbers that are raised to the 2nd power.
      * Example: calculateDifferenceOfSquares(3, 4) should return 7.
      * @param a The first number.
      * @param b The second number.
      * @return The absolute value of the difference
      */
     public static Double calculateDifferenceOfSquares(Double a, Double b) {
-        // Square the input numbers using the Math.pow() function
-
-        // Calculate the absolute difference using Math.abs() function
-
-        // Return the result
-        return null; // Replace null with the variable you used to store the result
+        Double doubleA = math.pow(a, 2);
+        Double doubleB = math.pow(b, 2);
+        Double result = math.abs(doubleA - doubleB);
```

Comment:
Be sure to always capitalize the names of classes:
```suggestion
        Double doubleA = Math.pow(a, 2);
        Double doubleB = Math.pow(b, 2);
        Double result = Math.abs(doubleA - doubleB);
```
Also, nice use of the `abs()` method to return the absolute (positive) value.